src/config.c

```diff
@@ -2165,6 +2159,14 @@ static int numericParseString(typeData data, sds value, const char **err, long l
             return 1;
     }
 
+    /* Attempt to parse as an octal number */
+    if (data.numeric.flags & OCTAL_CONFIG) {
+        char* eptr;
+        *res = strtoll(value, &eptr, 8);
+        if (*eptr == '\0')
+            return 1;
```

Comment:
A comment may help clarify how we detect non-octal digits.

Do we also need to check that we didn't get an empty string by checking eptr == value?

```suggestion
        *res = strtoll(value, &eptr, 8);
        if (*eptr == '\0')
            return 1; /* some non-octal digits remain */
        if (eptr == value)
            return 1; /* no octal digits found */
```

Reply:
I don't follow why we would do the second check? The man pages say we should be checking for errno here instead of the returned value, but it looks like within this page we don't consistently check it anyways.

Reply:
@madolson strtoll sets errno only on overflow/underflow and then it returns LLONG_MAX/LLONG_MIN. We check that implicitly since we check the valid range `0..0777` in this case.

Extra chars after the number is not an error to strtoll, but it is to us, and so is an empty string (I think it should be anyway). The way to check it is by endptr.

[Edit] Another way to detect an empty input string is to check `value[0] == '\0'` obviously.

![image](https://user-images.githubusercontent.com/273886/136337751-9e12c401-2188-46af-9283-6bc69c824529.png)

Reply:
Good catch! empty strings will result in 0

Seems we got this issue in few places like
```
This works:
./src/redis-server --oom-score-adj-values '' '' ''
which is same as
./src/redis-server --oom-score-adj-values 0 0 0

and this works
./src/redis-server --client-output-buffer-limit normal '' 0 0
```

Reply:
Ouch! So maybe we have to accept `""` as 0 for backward compatibility?

Reply:
..and in util memtoull(), observable with
`./src/redis-server --maxmemory-clients ''`